### PR TITLE
fix reporting invalid properties

### PR DESCRIPTION
**Pull request by adietish**
Signed-off-by: Andre Dietisheim <adietish@redhat.com>

### CODE REVIEW

**Comment by adietish** [comment]
@robstryker dare to review please?

File: framework/bundles/org.jboss.tools.rsp.server/src/main/java/org/jboss/tools/rsp/server/model/ServerModel.java [hunk 1]
```diff
@@ -235,41 +249,61 @@ private CreateServerResponse createServerUnprotected(String serverType, String i
 	}
 	
 	private IStatus validateAttributes(IServerType type, Map<String, Object> map) {
+		MultiStatus multiStatus = new MultiStatus(ServerCoreActivator.BUNDLE_ID, 0,
```
[the remark this answers is not in the document]
Comment (adietish) [reply framework/bundles/org.jboss.tools.rsp.server/src/main/java/org/jboss/tools/rsp/server/model/ServerModel.java]: the multistatus wont go over the wire. Your conversion util does not consider the child status. The purpose of the multistatus here is to collect the errors and return a merged status. Once the errors are collected they're extracted and only the keys are kept and sent. See here: https://github.com/redhat-developer/rsp-server/pull/169/files#diff-e1f6bbde631355b5bf4d80116fc8722cR236
```
if( !validAttributes.isOK()) { // merged status as calculated by the multistatus
	return new CreateServerResponse(StatusConverter.convert(validAttributes), // wont consider child status
	getInvalidAttributeKeys(validAttributes)); // we're keeping the keys of the invalid attributes
}
```

File: framework/bundles/org.jboss.tools.rsp.server/src/main/java/org/jboss/tools/rsp/server/model/ServerModel.java [hunk 2]
```diff
@@ -199,32 +212,33 @@ public CreateServerResponse createServer(String serverType, String id, Map<Strin
...
 			return new CreateServerResponse(StatusConverter.convert(s), null);
 		}
 	}
 	
-	private CreateServerResponse createServerUnprotected(String serverType, String id, Map<String, Object> attributes) 
-			throws CoreException {
-		if( servers.get(id) != null ) {
+
+	private CreateServerResponse createServerUnprotected(String serverType, String id, Map<String, Object> attributes) throws CoreException {
+		if( servers.containsKey(id)) {
```
[the remark this answers is not in the document]
Comment (adietish) [reply framework/bundles/org.jboss.tools.rsp.server/src/main/java/org/jboss/tools/rsp/server/model/ServerModel.java]: hum, but Map#put would throw if you put a null value?
```
     * @throws NullPointerException if the specified key or value is null
     *         and this map does not permit null keys or values
     */
    V put(K key, V value);
```

**Comment by adietish** [comment]
@robstryker addressed and corrected according to discussion in https://github.com/redhat-developer/rsp-server/pull/169#discussion_r234704972
I think that the other concerns are good points to consider but the impl behaves as they should.
Dare to review, please?

**Comment by robstryker** [comment]
+1

**Pull request merged**